app/controllers/concerns/idv/verify_info_concern.rb

```diff
@@ -414,5 +379,125 @@ def threatmetrix_failure_reason(success, result)
     def add_cost(token, transaction_id: nil)
       Db::SpCost::AddSpCost.call(current_sp, token, transaction_id: transaction_id)
     end
+
+    def log_verification_attempt(success:, failure_reason: nil)
+      pii_from_doc = pii || {}
+
+      attempts_api_tracker.idv_verification_submitted(
+        success: success,
+        document_state: pii_from_doc[:state],
+        document_number: pii_from_doc[:state_id_number],
+        document_issued: pii_from_doc[:state_id_issued],
+        document_expiration: pii_from_doc[:state_id_expiration],
+        first_name: pii_from_doc[:first_name],
+        last_name: pii_from_doc[:last_name],
+        date_of_birth: pii_from_doc[:dob],
+        address1: pii_from_doc[:address1],
+        address2: pii_from_doc[:address2],
+        social_security: idv_session.ssn,
+        failure_reason:,
+      )
+    end
+
+    VerificationFailures = Struct.new(
```

Comment:
this is probably the wrong name for this object, since it includes more than failures technically, but it's only being used to parse and handle failures. recommendations welcome!

Reply:
It's kind of a "result" since it's more than only failures, but that's already a pretty overloaded term in this context. I don't have a good suggestion 😅

Reply:
Yep, makes sense to me.